pkg/volume/plugins.go

```diff
@@ -131,26 +131,26 @@ type VolumePluginMgr struct {
 
 // Spec is an internal representation of a volume.  All API volume types translate to Spec.
 type Spec struct {
-	Name                   string
-	VolumeSource           api.VolumeSource
-	PersistentVolumeSource api.PersistentVolumeSource
-	ReadOnly               bool
+	Name             string
+	Volume           *api.Volume
```

Comment:
At the risk of bikeshedding, we could have a utility function that gave you the name.